test-broken-links.js

```diff
@@ -0,0 +1,72 @@
+const fs = require("fs");
+
+const WEBSITE_DIRECTORY = "https://legend.finos.org/";
+const DOC_DIRECTORY = "website/static/resource/studio/documentation/";
+
+var files = fs.readdirSync(DOC_DIRECTORY);
+
+for (let i = 0; i < files.length; i++) {
+  if (!files[i].endsWith(".json")) continue;
+  var jsonObject = JSON.parse(
+    fs.readFileSync(DOC_DIRECTORY + files[i], "utf8")
+  );
+
+  const keys = Object.keys(jsonObject.entries);
+  for (let i = 0; i < keys.length; i++) {
+    const key = keys[i];
+    if (jsonObject.entries[key].url == undefined) {
+      continue;
+    }
+    lookForUrl(jsonObject.entries[key].url);
+  }
+}
+
+async function lookForUrlWithAnchor(url) {
+  const locationOfFile = url
+    .substring(0, url.lastIndexOf("/"))
+    .replace(WEBSITE_DIRECTORY, "");
+
+  const filePathName =
+    "./website/build/" +
+    locationOfFile +
+    url.substring(url.lastIndexOf("/"), url.lastIndexOf("#")) +
+    ".html";
+
+  const anchorTag = url.substring(url.lastIndexOf("#"));
+
+  if (await fs.existsSync(filePathName)) {
+    await fs.readFile(filePathName, "UTF-8", (err, data) => {
+      if (!data) {
+        throw new Error("Broken link found " + filePathName);
```

Comment:
I think we could improve this script slightly by collecting the errors and throw in the end if the list is non-empty. If you make errors here, I suppose you would have to run multiple times to find out.